src/vmm/src/cpu_config/x86_64/cpuid/normalize.rs

```diff
@@ -349,47 +345,46 @@ impl super::Cpuid {
-                        // level Otherwise there are 2
+                        // domain; otherwise there are 2
                         set_range(&mut subleaf.result.ebx, 0..16, u32::from(cpus_per_core))
                             .map_err(ExtendedTopologyError::LogicalProcessors)?;
 
-                        set_range(&mut subleaf.result.ecx, 8..16, LEVEL_TYPE_THREAD)
-                            .map_err(ExtendedTopologyError::LevelType)?;
+                        // Skip setting 0 to ECX[7:0] since it's already reset to 0.
+
+                        // Set the domain type identification value for logical processor,
+                        set_range(&mut subleaf.result.ecx, 8..16, 1)
+                            .map_err(ExtendedTopologyError::DomainType)?;
                     }
-                    // Core Level Processor Topology; index = 1
+                    // Core domain
                     1 => {
                         set_range(&mut subleaf.result.eax, 0..5, LEAFBH_INDEX1_APICID)
                             .map_err(ExtendedTopologyError::ApicId)?;
 
+                        // Configure such that the next higher-scoped domain (i.e. socket) include
+                        // all logical processors.
                         set_range(&mut subleaf.result.ebx, 0..16, u32::from(cpu_count))
                             .map_err(ExtendedTopologyError::LogicalProcessors)?;
 
-                        // We expect here as this is an extremely rare case that is unlikely to ever
-                        // occur. It would require manual editing of the CPUID structure to push
-                        // more than 2^32 subleaves.
-                        let sub = index;
-                        set_range(&mut subleaf.result.ecx, 0..8, sub)
-                            .map_err(ExtendedTopologyError::LevelNumber)?;
-
-                        set_range(&mut subleaf.result.ecx, 8..16, LEVEL_TYPE_CORE)
-                            .map_err(ExtendedTopologyError::LevelType)?;
+                        // Setting the input ECX value (i.e. `index`)
+                        set_range(&mut subleaf.result.ecx, 0..8, index)
+                            .map_err(ExtendedTopologyError::InputEcx)?;
+
+                        // Set the domain type identification value for core.
+                        set_range(&mut subleaf.result.ecx, 8..16, 2)
+                            .map_err(ExtendedTopologyError::DomainType)?;
                     }
-                    // Core Level Processor Topology; index >=2
-                    // No other levels available; This should already be set correctly,
-                    // and it is added here as a "re-enforcement" in case we run on
-                    // different hardware
+                    // Unknown domains
                     _ => {
                         // We expect here as this is an extremely rare case that is unlikely to ever
```

Comment:
Mh, this comments doesn't parse. "We expect here"? Did there used to be a `.expect()` here, and this comment is outdated, or what do we mean here? (i realize its preexisting, so "idk" is a valid answer to resolve this comment haha)

Reply:
I don't know the context behind why we commented like that, but I'll elaborate more on that (e.g. in which case the code can reach the path!